scipy/integrate/_cubature.py

```diff
@@ -203,45 +204,55 @@ def cubature(f, a, b, rule="gk21", rtol=1e-05, atol=1e-08, max_subdivisions=1000
     subdivisions = 0
     success = True
 
-    while np.any(err > atol + rtol * np.abs(est)):
-        # region_k is the region with highest estimated error
-        region_k = heapq.heappop(regions)
+    with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
This `with` statement we want _outside_ the `cubature` function. It is probably expensive. Then you pass in either `executor` or `executor.map` to the `cubature`.